### DISCUSSION TITLE
Typography: Target inline code elements only

### DISCUSSION THREAD

**Discussion by aarondfrancis** [tailwindlabs/tailwindcss#12496] [Ideas]
I just wanted to leave this here in case anyone ever needs it. 

If you're trying to target `code` elements that are not wrapped in `pre` blocks with Tailwind Typography, you can add this variant to your `tailwind.config.js`.

```js
    plugins: [
        require('@tailwindcss/typography'),
        plugin(function ({addVariant}) {
          addVariant('prose-inline-code', '&.prose :where(:not(pre)>code):not(:where([class~="not-prose"] *))');
        })
    ],
```

You can use `prose-inline-code:{utility}` to target inline blocks now. 

I _think_ the selector is right, but it definitely works. I tried to mirror the default prose ones.

You can read more here: https://aaronfrancis.com/2023/tailwind-typography-inline-code-only

**Comment by thiagomajesk** [comment]
Lost more time than I'm willing to admit looking for a solution. Was trying to do this today and it was such a lifesaver 💪 (this is what I call timing). I wish this would be part of core utilities, since you cannot style `prose-code` without hitting on `pre > code` (usually not what you want). Thanks @aarondfrancis 👏 .

**Reply by aarondfrancis** [reply]
So glad it was helpful! <3

**Comment by adamwathan** [comment]
I think the fact that it currently targets code elements in pre elements is honestly a bug, haha. Feel free to make an issue on the typography repo 💀

**Comment by aarondfrancis** [comment]
Woah

**Comment by aarondfrancis** [comment]
For posterity, here's the issue: https://github.com/tailwindlabs/tailwindcss-typography/issues/329

**Comment by zenovak** [comment]
For those who copied the above snippet and having some tailwindcss intellisense issues or snippet not working

you need to also import the plugin 
```
/** @type {import('tailwindcss').Config} */
const plugin = require('tailwindcss/plugin')
 
module.exports = {
  plugins: [
    require('@tailwindcss/typography'),
 
    plugin(function ({addVariant}) {
      addVariant( 
        'prose-inline-code',
        '&.prose :where(:not(pre)>code):not(:where([class~="not-prose"] *))'
      );
    })
  ],
};
```

[docs](https://tailwindcss.com/docs/plugins)

**Comment by ace-of-aces** [comment]
For anyone wondering how to do this in Tailwind v4 CSS-only configuration, here's a translated custom variant:

```css
@custom-variant prose-inline-code {
  &.prose :where(:not(pre)>code):not(:where([class~="not-prose"] *)) {
    @slot;
  }
}
```
